uni/lib/model/entities/lecture.dart

```diff
@@ -51,19 +73,21 @@ class Lecture {
     String classNumber,
     int occurrId,
   ) {
-    final startTimeList = startTimeString.split(':');
-    final startTime = day.add(
-      Duration(
-        hours: int.parse(startTimeList[0]),
-        minutes: int.parse(startTimeList[1]),
-      ),
-    );
-    final endTime = startTime.add(Duration(minutes: 30 * blocks));
+    final startTimeHours = int.parse(startTimeString.substring(0, 2));
+    final startTimeMinutes = int.parse(startTimeString.substring(3, 5));
+    final endTimeHours =
+        (startTimeMinutes + (blocks * 30)) ~/ 60 + startTimeHours;
+    final endTimeMinutes = (startTimeMinutes + (blocks * 30)) % 60;
     return Lecture(
       subject,
       typeClass,
-      startTime,
-      endTime,
+      day.add(Duration(hours: startTimeHours, minutes: startTimeMinutes)),
+      day.add(
+        Duration(
+          hours: startTimeMinutes + endTimeHours,
+          minutes: startTimeMinutes + endTimeMinutes,
+        ),
+      ),
```

Comment:
Some git issues here, this is old code no longer on develop. Take a look at that